pkg/river/internal/rivertags/rivertags.go

```diff
@@ -174,7 +183,27 @@ func Get(ty reflect.Type) []Field {
 			usedLabelField = tf.Index
 		}
 
-		if fullName == "" && tf.Flags&FlagLabel == 0 /* (e.g., *not* a label) */ {
+		if tf.Flags&FlagSquash != 0 {
```

Comment:
I cleaned this up just a little bit. I'll revisit later for more clean-up though, it'd have to go through some major refactoring to be broken down more since there's function-wide state.